### ISSUE TITLE
`edgedb instance list` should still report local instances if the cloud is unreachable

### ISSUE THREAD

**Issue by zackelan** [edgedb/edgedb-cli#869]
- EdgeDB CLI Version: master on 2022-11-01 (`303ae71462463fa644893ed6ced03f6da3fa25ee`)

**repro**

configure at least 2 EdgeDB instances - one in the cloud, and one not in the cloud

```
$ edgedb instance list
┌────────┬────────────────┬────────────────────────────────────┬─────────────┬───────────┐
│ Kind   │ Name           │ Port                               │ Version     │ Status    │
├────────┼────────────────┼────────────────────────────────────┼─────────────┼───────────┤
│ remote │ edge25         │ localhost:15656                    │ 2.5+1965887 │ up        │
│ cloud  │ zackelan/test  │ nomad-node2.local-1.internal:27703 │ ?           │ available │
│ cloud  │ zackelan/test2 │ nomad-node2.local-1.internal:27499 │ ?           │ available │
└────────┴────────────────┴────────────────────────────────────┴─────────────┴───────────┘
```

cause the cloud to be inaccessible, such as by tearing down your local Nebula region

(this simulates a real-world scenario of EdgeDB Cloud actually being down, or simply an offline scenario where I'm developing on an airplane without wifi)

**results**

```
$ edgedb instance list
edgedb error: HTTP error: failed to lookup address information: Name or service not known
```

**expected results**

`instance list` should still tell me about my non-cloud instance, plus some indicator that my cloud instance isn't reachable

more broadly, if the cloud isn't reachable, all CLI operations involving local instances should still work

**nice to have**

maybe an explicit `--offline` command-line switch? in the case above, DNS resolution fails fast, so the `instance list` also fails fast.

if the connection attempt were to fail slowly instead (such as hitting an N second timeout trying to connect to the cloud endpoint) then each invocation of `instance list` will also fail slowly.

**Comment by tailhook** [comment]
Can you take a look #863 and say if it works fine for you?

**Comment by zackelan** [comment]
tested `1a35737dae9a584905fe026cc63be00cee2aebcd` (current head of `continue-on-cloud-timeout` branch) and got the same error:

```
$ edgedb instance list
edgedb error: HTTP error: failed to lookup address information: Name or service not known
```

it's not timing out in my case, the failure happens basically instantly (my `EDGEDB_CLOUD_BASE_URL` is `http://api.local-1.internal:5959` and with the Nebula region torn down, DNS resolution of that address fails immediately)

I figured out a trick to simulate the cloud API truly timing out (`ssh 10.56.1.2` then `sudo ufw enable`, make sure to disable again before logging out) and tested that behavior. it works, but 30 seconds is a looooong time to wait:

```
$ time edgedb instance list
Checking cloud and 1 remote instances...
Timed out getting cloud instances.
┌────────┬────────┬─────────────────┬─────────────┬────────┐
│ Kind   │ Name   │ Port            │ Version     │ Status │
├────────┼────────┼─────────────────┼─────────────┼────────┤
│ remote │ edge25 │ localhost:15656 │ 2.5+1965887 │ up     │
└────────┴────────┴─────────────────┴─────────────┴────────┘

________________________________________________________
Executed in   30.01 secs      fish           external
   usr time    3.75 millis  200.00 micros    3.55 millis
   sys time    5.57 millis  237.00 micros    5.33 millis
```

and while poking at this, I also found a useful way to cause the fail-fast behavior I'm seeing without needing to actually tear down your Nebula region, if it helps with repro:

```
$ sudo ip rule add blackhole to 10.56.1.2 dport 5959
$ time edgedb instance list
edgedb error: HTTP error: An error occured while creating a new object: Invalid argument (os error 22)

________________________________________________________
Executed in    4.68 millis    fish           external
   usr time    0.00 millis    0.00 micros    0.00 millis
   sys time    6.84 millis  320.00 micros    6.52 millis

$ sudo ip rule del blackhole to 10.56.1.2 dport 5959
$ edgedb instance list
┌────────┬───────────────┬────────────────────────────────────┬─────────────┬───────────┐
│ Kind   │ Name          │ Port                               │ Version     │ Status    │
├────────┼───────────────┼────────────────────────────────────┼─────────────┼───────────┤
│ remote │ edge25        │ localhost:15656                    │ 2.5+1965887 │ up        │
│ cloud  │ zackelan/test │ nomad-node0.local-1.internal:28904 │ ?           │ available │
└────────┴───────────────┴────────────────────────────────────┴─────────────┴───────────┘
```

**Comment by tailhook** [comment]
> it's not timing out in my case, the failure happens basically instantly

Ah, right. I've also commented about this in #863.

**Comment by fantix** [comment]
This is currently fixed in the latest `master` (tested on 0332a2c89a975f73dfbcdbf7af19e6bdc4ea1dfb), but note that the error will still be the only output if there is no other (local) instances found.
